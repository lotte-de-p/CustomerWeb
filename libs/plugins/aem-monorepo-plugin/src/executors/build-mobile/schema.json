{
  "type": "object",
  "properties": {
    "appSPath": {
      "type": "string",
      "description": "Path to the mfe app"
    },
    "skipGenerateI18n": {
      "type": "boolean",
      "description": "Option to skip the generate-i18n step",
      "default": false
    }
  },
  "required": ["appPath"]
}
