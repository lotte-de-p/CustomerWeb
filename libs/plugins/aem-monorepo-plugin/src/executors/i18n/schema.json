{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "appPath": {
      "type": "string",
      "description": "Path to the mfe app."
    },
    "prefixes": {
      "type": "string",
      "description": "Comma-separated list of prefixes to filter i18n keys."
    }
  },
  "required": ["appPath", "prefixes"]
}
