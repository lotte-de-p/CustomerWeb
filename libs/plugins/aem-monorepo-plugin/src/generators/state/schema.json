{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "State",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "project": {
      "type": "string",
      "description": "The project name to add the state to (typical the data-access project inside a domain)",
      "alias": "p",
      "x-prompt": "The project to add state to?"
    },
    "facade": {
      "type": "boolean",
      "default": true,
      "description": "Should a facade be generated",
      "alias": "f",
      "x-prompt": "Should a facade be generated?"
    }
  },
  "required": ["name", "project"]
}
