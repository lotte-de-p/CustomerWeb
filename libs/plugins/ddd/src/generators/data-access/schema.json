{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "dataAccessOptions",
  "type": "object",
  "properties": {
    "facade": {
      "type": "boolean",
      "default": true,
      "description": "Should a facade be generated",
      "alias": "f",
      "x-prompt": "Should a facade be generated?"
    },
    "leanIxName": {
      "type": "string",
      "x-prompt": "What is the LeanIX name of the MFE?",
      "description": "LeanIX name"
    },
    "mfeName": {
      "description": "MFE name",
      "type": "string",
      "x-prompt": {
        "message": "What is the name of the MFE?",
        "type": "list",
        "items": [
          {
            "value": "barrings-mfe",
            "label": "barrings-mfe"
          },
          {
            "value": "care-billing-mfe",
            "label": "care-billing-mfe"
          },
          {
            "value": "address-checker-mfe",
            "label": "address-checker-mfe"
          },
          {
            "value": "care-service-cases-mfe",
            "label": "care-service-cases-mfe"
          },
          {
            "value": "customer-registration-mfe",
            "label": "customer-registration-mfe"
          },
          {
            "value": "dashboard-mfe",
            "label": "dashboard-mfe"
          },
          {
            "value": "demo-mfe",
            "label": "demo-mfe"
          },
          {
            "value": "entertainment-mfe",
            "label": "entertainment-mfe"
          },
          {
            "value": "fleet-mfe",
            "label": "fleet-mfe"
          },
          {
            "value": "general-mfe",
            "label": "general-mfe"
          },
          {
            "value": "inbox-messages-mfe",
            "label": "inbox-messages-mfe"
          },
          {
            "value": "invoice-mfe",
            "label": "invoice-mfe"
          },
          {
            "value": "limits-mfe",
            "label": "limits-mfe"
          },
          {
            "value": "marketing-mfe",
            "label": "marketing-mfe"
          },
          {
            "value": "monorepo-storybook-mfe",
            "label": "monorepo-storybook-mfe"
          },
          {
            "value": "payment-mfe",
            "label": "payment-mfe"
          },
          {
            "value": "product-mfe",
            "label": "product-mfe"
          },
          {
            "value": "product-journey-mfe",
            "label": "product-journey-mfe"
          },
          {
            "value": "profile-management-mfe",
            "label": "profile-management-mfe"
          },
          {
            "value": "return-devices-mfe",
            "label": "return-devices-mfe"
          },
          {
            "value": "sales-mfe",
            "label": "sales-mfe"
          },
          {
            "value": "sim-management-mfe",
            "label": "sim-management-mfe"
          },
          {
            "value": "wink-mfe",
            "label": "wink-mfe"
          }
        ]
      }
    },
    "domain": {
      "type": "string",
      "description": "Domainname"
    }
  },
  "required": ["mfeName", "leanIxName"]
}
