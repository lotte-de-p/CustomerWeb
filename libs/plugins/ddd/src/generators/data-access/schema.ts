/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface DataAccessOptions {
  /**
   * Should a facade be generated
   */
  facade?: boolean;
  /**
   * LeanIX name
   */
  leanIxName: string;
  /**
   * MFE name
   */
  mfeName: string;
  /**
   * Domainname
   */
  domain?: string;
  [k: string]: unknown;
}
