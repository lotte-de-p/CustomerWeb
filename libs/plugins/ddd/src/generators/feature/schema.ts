/* eslint-disable */
/**
 * This file was automatically generated by json-schema-to-typescript.
 * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
 * and run json-schema-to-typescript to regenerate this file.
 */

export interface FeatureOptions {
  /**
   * Feature name
   */
  name: string;
  /**
   * Domain name
   */
  domain?: string;
  /**
   * LeanIX name
   */
  leanIxName: string;
  /**
   * MFE name
   */
  mfeName: string;
  /**
   * Add ngrx for the domain (entity required)
   */
  ngrx?: boolean;
  [k: string]: unknown;
}
