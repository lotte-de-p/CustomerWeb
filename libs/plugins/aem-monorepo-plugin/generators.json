{
  "$schema": "http://json-schema.org/schema",
  "name": "aem-monorepo-plugin",
  "version": "0.0.1",
  "generators": {
    "state": {
      "factory": "./src/generators/state/stateGenerator",
      "schema": "./src/generators/state/schema.json",
      "description": "state generator"
    }
  }
}
