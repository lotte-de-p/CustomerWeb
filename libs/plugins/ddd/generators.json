{
  "$schema": "http://json-schema.org/schema",
  "name": "ddd",
  "version": "0.0.1",
  "generators": {
    "ddd:feature": {
      "factory": "./src/generators/feature",
      "schema": "./src/generators/feature/schema.json",
      "description": "DDD Feature generator"
    },
    "ddd:ui": {
      "factory": "./src/generators/ui",
      "schema": "./src/generators/ui/schema.json",
      "description": "DDD ui generator"
    }
  }
}
