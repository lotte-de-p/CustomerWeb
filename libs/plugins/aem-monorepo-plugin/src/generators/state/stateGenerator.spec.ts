import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Tree, readProjectConfiguration } from '@nrwl/devkit';

import generator from './stateGenerator';
import { StateGeneratorSchema } from './schema';
import { Linter } from '@nx/linter';
import { libraryGenerator, UnitTestRunner } from '@nrwl/angular/generators';

describe('state generator', () => {
  let libTree: Tree;
  beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation(jest.fn);
    libTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  describe(`When no facade should be generated`, () => {
    it('should generate the ngrx state files', async () => {
      const options: StateGeneratorSchema = { name: 'test', project: 'test-lib', facade: false };
      await libraryGenerator(libTree, {
        linter: Linter.EsLint,
        name: 'test-lib',
        skipFormat: true,
        style: 'scss',
        skipTests: true,
        unitTestRunner: UnitTestRunner.None,
      });
      await generator(libTree, options);
      const config = readProjectConfiguration(libTree, 'test-lib');
      expect(config).toBeDefined();

      expect(libTree.read('libs/test-lib/src/lib/+state/test-ngrx.module.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.state.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.selectors.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.reducer.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.effects.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.effects.spec.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.actions.ts', 'utf-8')).toMatchSnapshot();

      expect(libTree.exists('libs/test-lib/src/lib/+state/test.facade.ts')).toBe(false);
      expect(libTree.exists('libs/test-lib/src/lib/+state/test.facade.spec.ts')).toBe(false);
    });

    it(`should throw an exception if the project can't be found`, async () => {
      const options: StateGeneratorSchema = { name: 'test', project: 'unexisting-lib', facade: false };
      await libraryGenerator(libTree, {
        linter: Linter.EsLint,
        name: 'test-lib',
        skipFormat: true,
        style: 'scss',
        skipTests: true,
        unitTestRunner: UnitTestRunner.None,
      });
      try {
        await generator(libTree, options);
        fail('An exception should have been thrown');
      } catch (error) {
        expect(error).toEqual(new Error(`No project configuration found for project 'unexisting-lib'`));
      }
    });
  });
  describe(`When a facade should be generated`, () => {
    it('should generate the ngrx state files', async () => {
      const options: StateGeneratorSchema = { name: 'test', project: 'test-lib', facade: true };
      await libraryGenerator(libTree, {
        linter: Linter.EsLint,
        name: 'test-lib',
        skipFormat: true,
        style: 'scss',
        skipTests: true,
        unitTestRunner: UnitTestRunner.None,
      });
      await generator(libTree, options);
      const config = readProjectConfiguration(libTree, 'test-lib');
      expect(config).toBeDefined();

      expect(libTree.read('libs/test-lib/src/lib/+state/test-ngrx.module.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.state.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.selectors.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.reducer.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.effects.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.effects.spec.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.actions.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.facade.ts', 'utf-8')).toMatchSnapshot();
      expect(libTree.read('libs/test-lib/src/lib/+state/test.facade.spec.ts', 'utf-8')).toMatchSnapshot();
    });

    it(`should throw an exception if the project can't be found`, async () => {
      const options: StateGeneratorSchema = { name: 'test', project: 'unexisting-lib', facade: true };
      await libraryGenerator(libTree, {
        linter: Linter.EsLint,
        name: 'test-lib',
        skipFormat: true,
        style: 'scss',
        skipTests: true,
        unitTestRunner: UnitTestRunner.None,
      });
      try {
        await generator(libTree, options);
        fail('An exception should have been thrown');
      } catch (error) {
        expect(error).toEqual(new Error(`No project configuration found for project 'unexisting-lib'`));
      }
    });
  });
});
