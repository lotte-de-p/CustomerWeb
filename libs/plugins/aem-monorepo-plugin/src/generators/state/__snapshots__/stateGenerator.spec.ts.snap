// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state generator When a facade should be generated should generate the ngrx state files 1`] = `
"import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { TestEffects } from './test.effects';
import { testFeature } from './test.reducer';

@NgModule({
  imports: [EffectsModule.forFeature([TestEffects]), StoreModule.forFeature(testFeature)],
})
export class TestNgrxModule {}
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 2`] = `
"export interface TestState {
}

export const defaultState: TestState = {
};
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 3`] = `
"import { createFeatureSelector } from '@ngrx/store';

import { TestState } from './test.state';

export const featureState = createFeatureSelector<TestState>('testFeature');
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 4`] = `
"import { Action, createFeature, createReducer } from '@ngrx/store';

import { TestState, defaultState } from './test.state';

export const reducer = createReducer(
  defaultState,
);

export function testReducer(state: TestState = defaultState, action: Action): TestState {
  return reducer(state, action);
}

export const testFeature = createFeature({
  name: 'testFeature',
  reducer: testReducer,
});
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 5`] = `
"import { Injectable } from '@angular/core';
import { Actions } from '@ngrx/effects';
import * as fromActions from './test.actions';

@Injectable({
  providedIn: 'root',
})
export class TestEffects {
  constructor(private readonly actions$: Actions) {}
}
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 6`] = `
"import { TestBed } from '@angular/core/testing';
import { provideMockActions } from '@ngrx/effects/testing';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs';
import { provideMockStore } from '@ngrx/store/testing';

import * as testActions from './test.actions';
import { TestEffects } from './test.effects';

describe('TestEffects', () => {
  let effects: TestEffects;
  let actions$: Observable<Action>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        TestEffects,
        provideMockStore(),
        provideMockActions(() => actions$),
      ],
    });

    effects = TestBed.inject(TestEffects);
  });

  it(\`should create\`, () => {
    expect(effects).toBeTruthy();
  });
});
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 7`] = `
"import { createAction, props } from '@ngrx/store';

export const exampleAction = createAction('[test-lib - test] example');
export const exampleSuccessAction = createAction(
  '[test-lib - test] example success response',
  props<{ message: string }>()
);
export const exampleErrorAction = createAction('[test-lib - test] example Error', props<{ error: Error }>());
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 8`] = `null`;

exports[`state generator When a facade should be generated should generate the ngrx state files 9`] = `null`;

exports[`state generator When a facade should be generated should generate the ngrx state files 10`] = `
"import { createFeatureSelector } from '@ngrx/store';

import { TestState } from './test.state';

export const featureState = createFeatureSelector<TestState>('testFeature');
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 11`] = `
"import { Action, createFeature, createReducer } from '@ngrx/store';

import { TestState, defaultState } from './test.state';

export const reducer = createReducer(
  defaultState,
);

export function testReducer(state: TestState = defaultState, action: Action): TestState {
  return reducer(state, action);
}

export const testFeature = createFeature({
  name: 'testFeature',
  reducer: testReducer,
});
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 12`] = `
"import { Injectable } from '@angular/core';
import { Actions } from '@ngrx/effects';
import * as fromActions from './test.actions';

@Injectable({
  providedIn: 'root',
})
export class TestEffects {
  constructor(private readonly actions$: Actions) {}
}
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 13`] = `
"import { TestBed } from '@angular/core/testing';
import { provideMockActions } from '@ngrx/effects/testing';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs';
import { provideMockStore } from '@ngrx/store/testing';

import * as testActions from './test.actions';
import { TestEffects } from './test.effects';

describe('TestEffects', () => {
  let effects: TestEffects;
  let actions$: Observable<Action>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        TestEffects,
        provideMockStore(),
        provideMockActions(() => actions$),
      ],
    });

    effects = TestBed.inject(TestEffects);
  });

  it(\`should create\`, () => {
    expect(effects).toBeTruthy();
  });
});
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 14`] = `
"import { createAction, props } from '@ngrx/store';

export const exampleAction = createAction('[test-lib - test] example');
export const exampleSuccessAction = createAction(
  '[test-lib - test] example success response',
  props<{ message: string }>()
);
export const exampleErrorAction = createAction('[test-lib - test] example Error', props<{ error: Error }>());
"
`;

exports[`state generator When a facade should be generated should generate the ngrx state files 15`] = `null`;

exports[`state generator When a facade should be generated should generate the ngrx state files 16`] = `null`;

exports[`state generator When no facade should be generated should generate the ngrx state files 1`] = `
"import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { TestEffects } from './test.effects';
import { testFeature } from './test.reducer';

@NgModule({
  imports: [EffectsModule.forFeature([TestEffects]), StoreModule.forFeature(testFeature)],
})
export class TestNgrxModule {}
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 2`] = `
"export interface TestState {
}

export const defaultState: TestState = {
};
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 3`] = `
"import { createFeatureSelector } from '@ngrx/store';

import { TestState } from './test.state';

export const featureState = createFeatureSelector<TestState>('testFeature');
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 4`] = `
"import { Action, createFeature, createReducer } from '@ngrx/store';

import { TestState, defaultState } from './test.state';

export const reducer = createReducer(
  defaultState,
);

export function testReducer(state: TestState = defaultState, action: Action): TestState {
  return reducer(state, action);
}

export const testFeature = createFeature({
  name: 'testFeature',
  reducer: testReducer,
});
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 5`] = `
"import { Injectable } from '@angular/core';
import { Actions } from '@ngrx/effects';
import * as fromActions from './test.actions';

@Injectable({
  providedIn: 'root',
})
export class TestEffects {
  constructor(private readonly actions$: Actions) {}
}
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 6`] = `
"import { TestBed } from '@angular/core/testing';
import { provideMockActions } from '@ngrx/effects/testing';
import { Action } from '@ngrx/store';
import { Observable } from 'rxjs';
import { provideMockStore } from '@ngrx/store/testing';

import * as testActions from './test.actions';
import { TestEffects } from './test.effects';

describe('TestEffects', () => {
  let effects: TestEffects;
  let actions$: Observable<Action>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        TestEffects,
        provideMockStore(),
        provideMockActions(() => actions$),
      ],
    });

    effects = TestBed.inject(TestEffects);
  });

  it(\`should create\`, () => {
    expect(effects).toBeTruthy();
  });
});
"
`;

exports[`state generator When no facade should be generated should generate the ngrx state files 7`] = `
"import { createAction, props } from '@ngrx/store';

export const exampleAction = createAction('[test-lib - test] example');
export const exampleSuccessAction = createAction(
  '[test-lib - test] example success response',
  props<{ message: string }>()
);
export const exampleErrorAction = createAction('[test-lib - test] example Error', props<{ error: Error }>());
"
`;
